frontend/ui/widget/qrwidget.lua

```diff
@@ -26,7 +26,18 @@ function QRWidget:init()
         logger.info("QRWidget: failed to generate QR code.")
         return
     else
-        local scale = Screen:scaleBySize(5)
+        local scale
+        if self.width then
+            if self.height then
+                scale = math.min(self.width, self.height)/#grid
+            else
+                scale = self.width/#grid
+            end
+        elseif self.height then
+            scale = self.height/#grid
+        else scale = 2
+        end
+        scale = Screen:scaleBySize(scale)
         local grid_size = scale * #grid
```

Comment:
Not sure I understand all that: self.width and self.height are absolute sizes in px.
So, then, your 'scale' is an absolute size too: no need to scaleBySize() then (possibly, on your emulator, scaleBySize does *1). Your `scale` here might better be named something else (square_size?)

Reply:
Perhaps I didn't understand well what scaleBySize() does, but it seems it at least rounds values : when I don't use it (or math.ceil instead), I get "parasite" lines between "pixels". Perhaps I should just use math.ceil instead ?

Reply:
Example without rounding:

![qr](https://user-images.githubusercontent.com/4775231/98054042-4507af00-1e3a-11eb-8edc-4f032e77b2b7.png)

Reply:
scaleBySize just multiplies by a static factor made out of the `Screen>DPI` value and the screen w/h.
It's just so that we can use values like 5 or 10, which, once scaleBySize() should work/look just magically the same on large screen devices.
(I don't really get how that formula is made out, I just know its *1 on my emulator, and *1.79 on my Kobo :)

Reply:
> I get "parasite" lines between "pixels". 

Looks like you're painting white squares on a black background. Better reverse that: create a BB, fill it white, paint black square.
And you might miss 1px (due to some rounding error?) in the squares you paint, so the parasite lines.